lib/ASRouterTriggerListeners.jsm

```diff
@@ -24,11 +24,28 @@ this.ASRouterTriggerListeners = new Map([
     _triggerHandler: null,
     _hosts: null,
 
+    /**
+     * Wait for browser startup to finish to avoid accessing uninitialized
+     * properties
+     */
+    async _checkStartupFinished(win) {
+      if (!win.gBrowserInit.delayedStartupFinished) {
+        await new Promise(resolve => {
+          let delayedStartupObserver = () => {
+            Services.obs.removeObserver(delayedStartupObserver, "browser-delayed-startup-finished");
```

Comment:
I see similar code in browser.js explicitly checking the message and target… maybe it's just being unnecessarily careful? https://searchfox.org/mozilla-central/rev/b4ebbe90ae4d0468fe6232bb4ce90699738c8125/browser/base/content/browser.js#1166-1171

Reply:
Given the source of the bug I think it's a good idea. Added the check.